lib/getWorkflow.js

```diff
@@ -57,16 +65,16 @@ const calculateEdges = (jobs) => {
         const dest = j;
 
         if (Array.isArray(job.requires)) {
-            const specialTriggers = job.requires.filter(name => name.charAt(0) === '~');
-            const normalTriggers = job.requires.filter(name => name.charAt(0) !== '~');
-            const isJoin = normalTriggers.length > 1;
+            const specialTriggers = new Set(job.requires.filter(name => name.charAt(0) === '~'));
+            const normalTriggers = new Set(job.requires.filter(name => name.charAt(0) !== '~'));
+            const isJoin = normalTriggers.size > 1;
 
             specialTriggers.forEach((src) => {
-                edges.push({ src, dest });
+                edges.push({ src: filterNodeName(src), dest });
             });
 
             normalTriggers.forEach((src) => {
-                const obj = { src, dest };
+                const obj = { src: filterNodeName(src), dest };
```

Comment:
normalTriggers do not have `~`. Why we still need to `filterNodeName`?

Reply:
fixed